tests/spread/plugins/craft-parts/matter-sdk/snapcraft.yaml

```diff
@@ -33,13 +33,14 @@ parts:
   lighting:
     plugin: nil
     after: [matter-sdk]
+    build-environment:
+      - MATTER_SDK_ENV: ../../matter-sdk/build/matter_sdk_env
     override-build: |
-      cd ../../matter-sdk/build/matter-sdk
+      # Source the Matter SDK environment variables
+      source $MATTER_SDK_ENV
 
-      source matter_sdk_env
-
-      # Build the lighting app
-      cd examples/lighting-app/linux
+      # Build the lighting app for snapcraft spread testing purposes
+      cd ../../matter-sdk/build/examples/lighting-app/linux
       gn gen out/build
```

Comment:
Sourced the Matter SDK environment variables above, but the build is lacking the necessary environment variables:
```
source $MATTER_SDK_ENV
:: ++ export CIPD_CACHE_DIR=/root/.cipd-cache-dir                                                                                                                                   
:: ++ CIPD_CACHE_DIR=/root/.cipd-cache-dir                                                                                                                                          
:: ++ export PATH=/root/parts/matter-sdk/build/third_party/pigweed/repo/out/host/host_tools:/root/parts/matter-sdk/build/.environment/pigweed-venv/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm:/root/parts/matter-sdk/build/.environment/cipd/packages/zap:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed:/root/parts/matter-sdk/build/.environment/cipd:/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                                      
:: ++ PATH=/root/parts/matter-sdk/build/third_party/pigweed/repo/out/host/host_tools:/root/parts/matter-sdk/build/.environment/pigweed-venv/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm:/root/parts/matter-sdk/build/.environment/cipd/packages/zap:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed:/root/parts/matter-sdk/build/.environment/cipd:/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                                             
:: ++ export PATH=/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                       
:: ++ PATH=/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                              
:: ++ export PW_ARM_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/arm                                                                                    
:: ++ PW_ARM_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/arm                                                                                           
:: ++ export PW_BRANDING_BANNER=/root/parts/matter-sdk/build/scripts/setup/banner.txt                                                                                               
:: ++ PW_BRANDING_BANNER=/root/parts/matter-sdk/build/scripts/setup/banner.txt                                                                                                      
:: ++ export PW_PACKAGE_ROOT=/root/parts/matter-sdk/build/.environment/packages                                                                                                     
:: ++ PW_PACKAGE_ROOT=/root/parts/matter-sdk/build/.environment/packages                                                                                                            
:: ++ export PW_PIGWEED_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed                                                                            
:: ++ PW_PIGWEED_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed                                                                                   
:: ++ export PW_PROJECT_ROOT=/root/parts/matter-sdk/build                                                                                                                           
:: ++ PW_PROJECT_ROOT=/root/parts/matter-sdk/build                                                                                                                                  
:: ++ export PW_PYTHON_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/python                                                                              
:: ++ PW_PYTHON_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/python                                                                                     
:: ++ export PW_ROOT=/root/parts/matter-sdk/build/third_party/pigweed/repo                                                                                                          
:: ++ PW_ROOT=/root/parts/matter-sdk/build/third_party/pigweed/repo                                                                                                                 
:: ++ export PW_ZAP_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/zap                                                                                    
:: ++ PW_ZAP_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/zap                                                                                           
:: ++ export VIRTUAL_ENV=/root/parts/matter-sdk/build/.environment/pigweed-venv                                                                                                     
:: ++ VIRTUAL_ENV=/root/parts/matter-sdk/build/.environment/pigweed-venv                                                                                                            
:: ++ export _PW_ACTUAL_ENVIRONMENT_ROOT=/root/parts/matter-sdk/build/.environment                                                                                                  
:: ++ _PW_ACTUAL_ENVIRONMENT_ROOT=/root/parts/matter-sdk/build/.environment                                                                                                         
:: ++ export _PW_ENVIRONMENT_CONFIG_FILE=/root/parts/matter-sdk/build/scripts/setup/environment.json                                                                                
:: ++ _PW_ENVIRONMENT_CONFIG_FILE=/root/parts/matter-sdk/build/scripts/setup/environment.json                                                                                       
:: ++ export _PW_ROSETTA=0                                                                                                                                                          
:: ++ _PW_ROSETTA=0                                                                                                                                                                 
:: + cd ../../matter-sdk/build/examples/lighting-app/linux 
:: + gn gen out/build                                                                                                                                                               
:: ERROR at //third_party/connectedhomeip/build/config/linux/pkg_config.gni:116:17: Script returned non-zero exit code.                                                             
::     pkgresult = exec_script(pkg_config_script, args, "value")                                                                                                                    
::                 ^----------                                                                                                                                                      
:: Current dir: /root/parts/matter-sdk/build/examples/lighting-app/linux/out/build/                                                                                                 
:: Command: python /root/parts/matter-sdk/build/examples/lighting-app/linux/third_party/connectedhomeip/build/config/linux/pkg-config.py openssl                                    
:: Returned 127.                                                                                                                                                                    
:: See //third_party/connectedhomeip/src/crypto/BUILD.gn:79:3: whence it was called.                                                                                                
::   pkg_config("openssl_config") {                                                                                                                                                 
::   ^-----------------------------                                                                                                                                                 
:: See //third_party/connectedhomeip/src/lib/BUILD.gn:22:5: which caused the file to be included.                                                                                   
::     "${chip_root}/src/crypto",                                                                                                                                                   
::     ^------------------------                                                                                                                                                    
'override-build' in part 'lighting' failed with code 1.
```

Reply:
The potential solution could be to export $PATH, which is the directory that contains executables related to Pigweed, in the Matter plugin.

Reply:
> Here are the differences in environment variables before and after activation:
```
OLDPWD=/root/parts/lighting/build                                                                                                                                
PATH=/root/parts/matter-sdk/build/third_party/pigweed/repo/out/host/host_tools:/root/parts/matter-sdk/build/.environment/pigweed-venv/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm:/root/parts/matter-sdk/build/.environment/cipd/packages/zap:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed:/root/parts/matter-sdk/build/.environment/cipd:/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                                                
PATH=/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                 PWD=/root/parts/lighting/build                                                                                                                                                   
PWD=/root/parts/matter-sdk/build
```
I doubt that PWD changes because of the activation. That is a shell built in variable updated when changing directory. Sourcing it from the file could have unexpected results.

For the PATH, it looks like the following gets prepended to it: `/root/parts/matter-sdk/build/third_party/pigweed/repo/out/host/host_tools:/root/parts/matter-sdk/build/.environment/pigweed-venv/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm:/root/parts/matter-sdk/build/.environment/cipd/packages/zap:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed:/root/parts/matter-sdk/build/.environment/cipd`, along with the separator.

Reply:
> Sourced the Matter SDK environment variables above, but the build is lacking the necessary environment variables:
> 
> ```
> Building lighting                                                                                                                                                                   
> :: + source ../../matter-sdk/build/matter_sdk_env  
> :: ++ export CIPD_CACHE_DIR=/root/.cipd-cache-dir                                                                                                                                   
> :: ++ CIPD_CACHE_DIR=/root/.cipd-cache-dir                                                                                                                                          
> :: ++ export PATH=/root/parts/matter-sdk/build/third_party/pigweed/repo/out/host/host_tools:/root/parts/matter-sdk/build/.environment/pigweed-venv/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm:/root/parts/matter-sdk/build/.environment/cipd/packages/zap:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed:/root/parts/matter-sdk/build/.environment/cipd:/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                                      
> :: ++ PATH=/root/parts/matter-sdk/build/third_party/pigweed/repo/out/host/host_tools:/root/parts/matter-sdk/build/.environment/pigweed-venv/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/arm:/root/parts/matter-sdk/build/.environment/cipd/packages/zap:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed/bin:/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed:/root/parts/matter-sdk/build/.environment/cipd:/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                                             
> :: ++ export PATH=/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                       
> :: ++ PATH=/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                              
> :: ++ export PW_ARM_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/arm                                                                                    
> :: ++ PW_ARM_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/arm                                                                                           
> :: ++ export PW_BRANDING_BANNER=/root/parts/matter-sdk/build/scripts/setup/banner.txt                                                                                               
> :: ++ PW_BRANDING_BANNER=/root/parts/matter-sdk/build/scripts/setup/banner.txt                                                                                                      
> :: ++ export PW_PACKAGE_ROOT=/root/parts/matter-sdk/build/.environment/packages                                                                                                     
> :: ++ PW_PACKAGE_ROOT=/root/parts/matter-sdk/build/.environment/packages                                                                                                            
> :: ++ export PW_PIGWEED_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed                                                                            
> :: ++ PW_PIGWEED_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/pigweed                                                                                   
> :: ++ export PW_PROJECT_ROOT=/root/parts/matter-sdk/build                                                                                                                           
> :: ++ PW_PROJECT_ROOT=/root/parts/matter-sdk/build                                                                                                                                  
> :: ++ export PW_PYTHON_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/python                                                                              
> :: ++ PW_PYTHON_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/python                                                                                     
> :: ++ export PW_ROOT=/root/parts/matter-sdk/build/third_party/pigweed/repo                                                                                                          
> :: ++ PW_ROOT=/root/parts/matter-sdk/build/third_party/pigweed/repo                                                                                                                 
> :: ++ export PW_ZAP_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/zap                                                                                    
> :: ++ PW_ZAP_CIPD_INSTALL_DIR=/root/parts/matter-sdk/build/.environment/cipd/packages/zap                                                                                           
> :: ++ export VIRTUAL_ENV=/root/parts/matter-sdk/build/.environment/pigweed-venv                                                                                                     
> :: ++ VIRTUAL_ENV=/root/parts/matter-sdk/build/.environment/pigweed-venv                                                                                                            
> :: ++ export _PW_ACTUAL_ENVIRONMENT_ROOT=/root/parts/matter-sdk/build/.environment                                                                                                  
> :: ++ _PW_ACTUAL_ENVIRONMENT_ROOT=/root/parts/matter-sdk/build/.environment                                                                                                         
> :: ++ export _PW_ENVIRONMENT_CONFIG_FILE=/root/parts/matter-sdk/build/scripts/setup/environment.json                                                                                
> :: ++ _PW_ENVIRONMENT_CONFIG_FILE=/root/parts/matter-sdk/build/scripts/setup/environment.json                                                                                       
> :: ++ export _PW_ROSETTA=0                                                                                                                                                          
> :: ++ _PW_ROSETTA=0                                                                                                                                                                 
> :: + cd ../../matter-sdk/build/examples/lighting-app/linux 
> :: + gn gen out/build                                                                                                                                                               
> :: ERROR at //third_party/connectedhomeip/build/config/linux/pkg_config.gni:116:17: Script returned non-zero exit code.                                                             
> ::     pkgresult = exec_script(pkg_config_script, args, "value")                                                                                                                    
> ::                 ^----------                                                                                                                                                      
> :: Current dir: /root/parts/matter-sdk/build/examples/lighting-app/linux/out/build/                                                                                                 
> :: Command: python /root/parts/matter-sdk/build/examples/lighting-app/linux/third_party/connectedhomeip/build/config/linux/pkg-config.py openssl                                    
> :: Returned 127.                                                                                                                                                                    
> :: See //third_party/connectedhomeip/src/crypto/BUILD.gn:79:3: whence it was called.                                                                                                
> ::   pkg_config("openssl_config") {                                                                                                                                                 
> ::   ^-----------------------------                                                                                                                                                 
> :: See //third_party/connectedhomeip/src/lib/BUILD.gn:22:5: which caused the file to be included.                                                                                   
> ::     "${chip_root}/src/crypto",                                                                                                                                                   
> ::     ^------------------------                                                                                                                                                    
> 'override-build' in part 'lighting' failed with code 1.
> ```

Isn't this because PATH has been duplicated? The second definition excludes the required paths.

Reply:
Yes, the first PATH is what the override-build needs.